binderhub/repoproviders.py

```diff
@@ -208,11 +224,24 @@ def get_resolved_ref(self):
         self.record_id = r.effective_url.rsplit("/", maxsplit=1)[1]
         return self.record_id
 
+    async def get_resolved_spec(self):
+        if not hasattr(self, 'record_id'):
+            self.record_id = await self.get_resolved_ref()
+        # zenodo registers a DOI which represents all versions.
+        # and it always resolves to latest version
+        # for that case, we have to replace it with record id
+        resolved_spec = self.spec.split("zenodo")[0] + "zenodo." + self.record_id
```

Comment:
Ahh! This creates a new DOI from the record ID! Maybe it is too early on Friday morning but I had to read this a few times to realise it :)

Reply:
i think my comment was also not clear. hopefully it is better now and feel free to change it :)